APE_specutil.rst

```diff
@@ -0,0 +1,481 @@
+information for a multi-dimensional spectrum. Conventions like the attribute
+names used in this class (where sensible) will then carry over to such future
+developments.
+
+Analysis tools to be included in specutils
+++++++++++++++++++++++++++++++++++++++++++
+
+In contrast to the above, much of the functionality needed to work with
+spectroscopy *operates on* spectra, rather than being an intrinsic part of the
+data structure.  `specutils` will contain standard tools of this sort, although
+the exact definition of “standard” (and therefore implemented in the
+`specutils` package) will need to be determined on a case-by-case basis. Some
+of the common functionality that has been identified as important and general
+enough for inclusion in `specutils` includes :
+
+* Continuum subtraction or normalization
+* Identification of spectral lines or other features
+* Measuring features in the spectra including fitting models, calculating line centroids/shapes, equivalent widths, and fluxes
+* Deblending of lines
+* Measuring velocities for the spectra including cross-correlation
+
+This should *not* be taken as an exhaustive list, however.  Additional
+functionality that meets the above guidelines will be implemented or accepted
+via PRs to `specutils` if of sufficient general interest.
+
+The next steps: specreduce and specviz
+--------------------------------------
+This APE is not intended to specify the full details of the ``specviz`` and
+``specreduce`` packages.  They will develop separately following the
+standard process for developing astropy packages, but this section
+outlines *example* functionality expected for these packages.
+
+``specreduce``:
+* Methods for extracting a 1D spectrum from a 2D image of a dispersed spectrum, including boxcar and “optimal” extraction
+* Wavelength calibration
+* Removal of sky features
+* Flux calibration of spectra
+* Integration with (but *not* duplication of) 2D image-level reduction steps from packages like `ccdproc <https://github.com/astropy/ccdproc>`_
+
+``specviz``:
+* Display a spectrum
+* Explore the spectra by zooming in on features or moving to different areas.
+*
+* Do analysis like line flux measurements (using algorithms provided in `specutils`) or line fitting
+* Overlay spectral line lists (including redshift offsets if relevant)
+
+
+
+Branches and pull requests
+--------------------------
+N/A
+
+
+Implementation
+--------------
+
+`specutils <https://github.com/astropy/specutils>`_ already exists in its
+github repository, and *part* of the `Spectrum1D` class in this APE has been 
+implemented there, but  will need to be completed if this APE is accepted.
+`specreduce <https://github.com/crawfordsm/specreduce>`_
+also has an implementation, which can be adapted to use the framework described
+here.  Similarly, visualization via the `specviz
+<https://github.com/spacetelescope/specreduce>`_ package will do the same.  Of
```

Comment:
This link points to `specreduce` and not `specviz` (https://github.com/spacetelescope/specviz).